eng/DotNetBuild.props

```diff
@@ -122,6 +122,8 @@
       <InnerBuildArgs Condition="'$(NetCoreAppToolCurrentVersion)' != ''">$(InnerBuildArgs) /p:NetCoreAppToolCurrentVersion=$(NetCoreAppToolCurrentVersion)</InnerBuildArgs>
 
       <InnerBuildArgs Condition="'$(EnableDefaultRidSpecificArtifacts)' != ''">$(InnerBuildArgs) /p:EnableDefaultRidSpecificArtifacts=$(EnableDefaultRidSpecificArtifacts)</InnerBuildArgs>
+
+      <InnerBuildArgs Condition="'$(TargetOS)' == 'ios' or '$(TargetOS)' == 'maccatalyst' or '$(TargetOS)' == 'tvos'">$(InnerBuildArgs) /p:BuildDarwinFrameworks=true</InnerBuildArgs>
```

Comment:
I don't have any context on this property. Can someone please explain why this isn't the default when targeting these OSs?

Reply:
that's a great question actually, looks like Zoltan asked for this in https://github.com/dotnet/runtime/pull/53370#discussion_r640886206 but I don't see a reason why it needs to be a flag still, the build perf impact for local builds should be negligible.